### PR TITLE
Advanced Darkness (wizard event)

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
The station is plunged into a terrible darkness for ~1 minute. Lights do not work on these shadows, you will have limited vision for the duration.

:cl: Kor
add: Added a new wizard event, Advanced Darkness.
/:cl:

### CODE REVIEW

**Comment by Anonus** [comment]
like normal darkness, except you can't magic missile it

**Comment by daman997** [comment]
https://www.youtube.com/watch?v=XjtkWZ1uCXo :+1:

**Comment by duncathan** [comment]
:+1: I'm glad you keep using your stuff to add fun features; it's good for the codebase as a whole
Keep up the good work kor

**Comment by PKPenguin321** [comment]
Itd be neat if slings could get something like this

File: code/modules/mining/lavaland/weather_types.dm [hunk 1]
```diff
@@ -28,7 +28,32 @@
...
 
+/datum/weather/advanced_darkness
+	name = "advanced darkness"
+	start_up_time = 100 //10 seconds
+	start_up_message = "The lights begin to dim...is power going out?"
+	duration_lower = 45
+	duration_upper = 60 //1 minute
+	duration_message = "This isn't regular darkness...this is advanced darkness!"
+	wind_down = 100 // 10 seconds
+	wind_down_message = "The darkness is receding. Thank god."
```
Comment (Alek2ander) [inline code/modules/mining/lavaland/weather_types.dm]: Thank [ticker.Bible_deity_name]

**Comment by KorPhaeron** [comment]
@duncathan 

I'm not sure how to respond to kind words please cease

@Alek2ander 

Good idea

@daman997 

I should probably get the quote correct

**Comment by tedward1337** [comment]
This isn't just REGULAR darkness
this is ADVANCED darkness

File: code/modules/mining/lavaland/weather_types.dm [hunk 2]
```diff
@@ -28,7 +28,32 @@
 
 
 
+/datum/weather/advanced_darkness
+	name = "advanced darkness"
+	start_up_time = 100 //10 seconds
+	start_up_message = "The lights begin to dim...is power going out?"
```
Comment (tkdrg) [inline code/modules/mining/lavaland/weather_types.dm]: space after the "..."

File: code/modules/mining/lavaland/weather_types.dm [hunk 3]
```diff
@@ -28,7 +28,32 @@
...
+	duration_lower = 45
+	duration_upper = 60 //1 minute
+	duration_message = "This isn't average everyday darkness...this is advanced darkness!"
```
Comment (tkdrg) [inline code/modules/mining/lavaland/weather_types.dm]: Same as above

File: code/modules/events/wizard/advanced_darkness.dm [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+	typepath = /datum/round_event/wizard/darkness
+	max_occurrences = 2
+	earliest_start = 0
+
+/datum/round_event/wizard/darkness
+
+	endWhen = 10
+
+/datum/round_event/wizard/darkness/start()
+	var/datum/weather/advanced_darkness/darkness = new /datum/weather/advanced_darkness
```
Comment (tkdrg) [inline code/modules/events/wizard/advanced_darkness.dm]: don't need the path here, just new is enough

File: code/modules/events/wizard/advanced_darkness.dm [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/datum/round_event_control/wizard/darkness
+	name = "Advanced Darkness"
+	weight = 2
+	typepath = /datum/round_event/wizard/darkness
+	max_occurrences = 2
+	earliest_start = 0
+
+/datum/round_event/wizard/darkness
+
```
Comment (tkdrg) [inline code/modules/events/wizard/advanced_darkness.dm]: why the random newline

**Comment by tkdrg** [comment]
Added a changelog here too.

**Comment by KorPhaeron** [comment]
@tkdrg

Wasnt ready to be merged aaaa

**Comment by tedward1337** [comment]
lmao

**Pull request merged**